File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // Get user input dollars
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    // Convert dollars to integer
    int cents = round(dollars * 100);

    int coins = 0;

    // Calculate minimum number of coins
    while (cents > 0)
    {
        // Greedy Algorithms
        if ((cents - 25) >= 0)
        {
            cents -= 25;
        }
        else if ((cents - 10) >= 0)
        {
            cents -= 10;
        }
        else if ((cents - 5) >= 0)
        {
            cents -= 5;
        }
        else if ((cents - 1) >= 0)
        {
            cents -= 1;
        }

        coins++;
    }

    printf("%i\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_float


def main():
    # Get user input dollars
    while True:
        dollars = get_float("Change owed: ")
        if dollars >= 0:
            break

    # Convert dollars to integer
    cents = round(dollars * 100)
    coins = 0

    # Calculate minimum number of coins
    while cents != 0:
        # Greedy Algorithms
        if (cents - 25) >= 0:
            cents -= 25
        elif (cents - 10) >= 0:
            cents -= 10
        elif (cents - 5) >= 0:
            cents -= 5
        elif (cents - 1) >= 0:
            cents -= 1

        coins += 1

    print(coins)


if __name__ == "__main__":
    main()